langchain/src/chat_models/openai.ts

```diff
@@ -418,58 +408,9 @@ export class ChatOpenAI
       // eslint-disable-next-line no-void
       void runManager?.handleLLMNewToken(generationChunk.text ?? "");
     }
-  }
-
-  startStream(
-    request: CreateChatCompletionRequest,
-    options?: StreamingAxiosConfiguration
-  ) {
-    let done = false;
-    const stream = new TransformStream();
-    const writer = stream.writable.getWriter();
-    const iterable = readableStreamToAsyncIterable(stream.readable);
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    let err: any;
-    this.completionWithRetry(request, {
-      ...options,
-      adapter: fetchAdapter, // default adapter doesn't do streaming
-      responseType: "stream",
-      onmessage: (event) => {
-        if (done) return;
-        if (event.data?.trim?.() === "[DONE]") {
-          done = true;
-          // eslint-disable-next-line no-void
-          void writer.close();
-        } else {
-          const data = JSON.parse(event.data);
-          if (data.error) {
-            done = true;
-            throw data.error;
-          }
-          // eslint-disable-next-line no-void
-          void writer.write(event.data);
-        }
-      },
-    }).catch((error) => {
-      if (!done) {
-        err = error;
-        done = true;
-        // eslint-disable-next-line no-void
-        void writer.close();
-      }
-    });
-    return {
-      async next() {
-        const chunk = await iterable.next();
-        if (err) {
-          throw err;
-        }
-        return chunk;
-      },
-      [Symbol.asyncIterator]() {
-        return this;
-      },
-    };
+    if (options.signal?.aborted) {
```

Comment:
CC @rattrayalex if the user aborts a stream, the SDK no longer throws an error. This is different from `fetch`'s behavior

Reply:
Yes, that's intentional – other than backcompat, does that cause problems / is there a reason it's not desirable behavior?

BTW, you likely want to define an `AbortError` class and throw it below – people identify these by checking that the `.name` of the error is `AbortError`.

Reply:
No, we can handle it this way - convention across other LLMs is to throw because they're fetch based so would like to keep it standard for us but no worries on your end then.